### PR TITLE
chore(proxy): crush build times

### PR DESCRIPTION
**Pull request by xla**
By boxing all our filters.

### CODE REVIEW

**Comment by rudolfs** [comment]
I wonder what the run-time and debugging implications of all the boxing are.

If we bump into any issues, we could also do the boxing dynamically for debug builds via a macro like in this PR: https://github.com/rs-ipfs/rust-ipfs/commit/ae3306686209afa5911b1ad02170c1ac3bacda7c (found [via](https://github.com/seanmonstar/warp/issues/507#issuecomment-615974062)).

**Comment by cloudhead** [comment]
> I wonder what the run-time and debugging implications of all the boxing are.

I'll buy everyone in the office a donut if it makes a difference!

**Comment by xla** [comment]
> I wonder what the run-time and debugging implications of all the boxing are.

The runtime implications are an increase in vtable lookups, we won't see a difference.

> If we bump into any issues, we could also do the boxing dynamically for debug builds via a macro like in this PR: [rs-ipfs/rust-ipfs@ae33066](https://github.com/rs-ipfs/rust-ipfs/commit/ae3306686209afa5911b1ad02170c1ac3bacda7c) (found [via](https://github.com/seanmonstar/warp/issues/507#issuecomment-615974062)).

I applied the macro instead as it isolates the change better, thanks for the reminder.

**Comment by cloudhead** [comment]
This is great, what kind of build time reduction are we looking at?

I would be in favor of always boxing, as it may reduce binary size quite a bit too (we should check).

**Comment by xla** [comment]
> This is great, what kind of build time reduction are we looking at?

After cargo clean, from 11min to 2:20. Incremental builds of the proxy binary from 7/8mins down to 20s.

**Comment by NunoAlexandre** [comment]
Super awesome!

File: proxy/src/http.rs [hunk 1]
```diff
@@ -21,6 +21,23 @@ mod source;
 mod transaction;
 mod user;
 
+/// Helper to combine the multiple filters together with Filter::or, possibly boxing the types in
```
Comment (NunoAlexandre) [inline proxy/src/http.rs:24]:
```suggestion
/// Helper to combine the multiple filters together with Filter::or, boxing the types in
```
?

Comment (rudolfs) [reply proxy/src/http.rs:24]: Whoops, I got too excited about the PR and forgot to apply this change : |

**Pull request merged**